src/patterns.md

```diff
@@ -111,6 +111,10 @@ if let (a, 3) = (1, 2) {           // "(a, 3)" is refutable, and will not match
 }
 ```
 
+Using an irrefutable pattern in `match`, `if let` or `while let` expressions produces a compiler warning. E.g. `if let (x, y) = (1, 2) { ... }` results in this compiler message `#[warn(irrefutable_let_patterns)]`.
```

Comment:
In general, the reference does not cover `rustc` lints, so I don't think it is appropriate to add this.

Reply:
Eric, in that case you should probably close issue #799 making this PR redundant.